tests/run_tests/apps_run_test.py

```diff
@@ -21,54 +24,152 @@
 # import pytest
 # pytest.skip("eliminating conflicting test.", allow_module_level=True)
 
-project = "./FlinFlon.geoh5"
-
-geoh5 = Workspace(project)
+PROJECT = "./FlinFlon.geoh5"
+GEOH5 = Workspace(PROJECT)
 
-project_dcip = "./FlinFlon_dcip.geoh5"
 
+def test_calculator(tmp_path):
+    temp_workspace = path.join(tmp_path, "contour.geoh5")
+    with Workspace(temp_workspace) as workspace:
+        GEOH5.get_entity("geochem")[0].copy(parent=workspace)
 
-def test_calculator():
-    app = Calculator(h5file=project)
+    app = Calculator(h5file=temp_workspace)
     app.trigger.click()
 
+    files = os.listdir(path.join(tmp_path, "Temp"))
+    with Workspace(path.join(tmp_path, "Temp", files[0])) as workspace:
+        output = workspace.get_entity("NewChannel")[0]
+        assert output.values.shape[0] == 4438, "Change in output. Need to verify."
+
 
-def test_coordinate_transformation():
-    app = CoordinateTransformation(h5file=project)
+def test_coordinate_transformation(tmp_path):
+    temp_workspace = path.join(tmp_path, "contour.geoh5")
+    with Workspace(temp_workspace) as workspace:
+        GEOH5.get_entity("Gravity_Magnetics_drape60m")[0].copy(parent=workspace)
+        GEOH5.get_entity("Data_TEM_pseudo3D")[0].copy(parent=workspace)
+
+    app = CoordinateTransformation(h5file=temp_workspace)
     app.trigger.click()
 
+    files = os.listdir(path.join(tmp_path, "Temp"))
+    with Workspace(path.join(tmp_path, "Temp", files[0])) as workspace:
+        assert len(workspace.objects) == 2, "Coordinate transform failed."
+
 
-def test_contour_values():
-    app = ContourValues(h5file=project, plot_result=False)
+def test_contour_values(tmp_path):
+    temp_workspace = path.join(tmp_path, "contour.geoh5")
+    with Workspace(temp_workspace) as workspace:
+        GEOH5.get_entity(uuid.UUID("{538a7eb1-2218-4bec-98cc-0a759aa0ef4f}"))[0].copy(
+            parent=workspace
+        )
+
+    app = ContourValues(h5file=temp_workspace, plot_result=False)
     app.trigger.click()
 
+    files = os.listdir(path.join(tmp_path, "Temp"))
+    with Workspace(path.join(tmp_path, "Temp", files[0])) as workspace:
+        output = workspace.get_entity("Airborne_TMI")[0]
+        assert output.n_vertices == 2740, "Change in output. Need to verify."
```

Comment:
Diggin the extra asserts - much stronger than the plain run()!

Reply:
indeed. The turns out that calling .run() didn't garantee that it actually ran because the ipywidgets appears to try/except pass if there is an error.

Reply:
Changed along with GEOPY-533